loomio/loomio
# Improved group privacy settings

**CR** : :white_check_mark: (mix) 

:x: : **QA** 

In this commit we have
- renamed group.viewable_by = 'members' to group.privacy = 'secret' (and changed 'everyone' to 'public')
- removed the group privacy drop-down toggle on the privacy icon
- combined the group setup form and group edit form

So if you're QA this, you should make sure to test:
- adding subgroups
- making new groups
- updating existing groups
- changing privacy settings
### QA Checklist

> _test core functionality of feature:_
> checked migration of existing groups :white_check_mark:
> 
> **updating privacy settings**
> for subgroups: 
> 
> > to public, checked signed in/out :white_check_mark: 
> > to secret setting, checked signed in/out :white_check_mark: 
> > to secret+parent_group_members, checked... :white_check_mark:
> 
> _for parent groups_: dito :white_check_mark: 
> 
> **creating a subgroup**
> 
> > in a public parent group :white_check_mark: 
> > in a secret parent group :x: - i was able to create a public group in a private parent group
> 
> **making new groups**
> 
> > private :white_check_mark: 
> > public :white_check_mark: 
> > :question: :warning: the default checkbox selected is private ... is this right ?
> 
> i didn't check invite_to_group security. not relevant :question:  

notes: 
- not being able to click on the lock to get to privacy settings is annoying, maybe make it a link
- when i set up a group, i get an email saying 'thanks for trying Loomio' and showing me how to invite people. given I'm already an admin and have started some serious groups this is kinda dumb.

> logged out functionality :white_check_mark: 
> 
> test in IE 8,9 :ok: - **NA** 
> test in production environment :ok: - nope

---

:question: : **CO signoff** 
:white_check_mark: : **DO signoff**

:question: : **PO signoff**
